make/common/JdkNativeCompilation.gmk

```diff
@@ -187,6 +187,11 @@ define SetupJdkLibraryBody
   # Add the module specific java header dir
   $1_SRC_HEADER_FLAGS += $$(addprefix -I, $$(call GetJavaHeaderDir, $$(MODULE)))
 
+  $1_JDK_LIBS += $$($1_JDK_LIBS_$$(OPENJDK_TARGET_OS))
```

Comment:
Should these follow LIBS and pick up more specific target options as well? Not a request, just thinking out loud

Reply:
I'm not sure what targets you are thinking of here. This is one of the cases where the distinction between OS and toolchain is not entirely clear. I'm treating them as synonyms. There are two reasons two separate target OS here:

1) syntax in linker. The microsoft linker uses `jvm.lib`, all others `-ljvm`.

2) different needs in the source code. When there are vastly different implementations for a lib on different OSes, we can have a situation where e.g. the macOS port needs libjava, but the other platforms don't.

Issue 1) will be resolved by the upcoming JDK_LIBS rewrite. Issue 2 is inherited from the limitations in the source code, and since that source code is structured on a per-OS basis, we will only need to have JDK_LIBS options per-OS.